Firestore/core/src/firebase/firestore/core/firestore_client.mm

```diff
@@ -119,15 +125,9 @@ new FirestoreClient(database_info, std::move(credentials_provider),
   shared_client->credentials_provider_->SetCredentialChangeListener(
       credential_change_listener);
 
-  // Defer initialization until we get the current user from the
-  // credential_change_listener. This is guaranteed to be synchronously
-  // dispatched onto our worker queue, so we will be initialized before any
-  // subsequently queued work runs.
-  shared_client->worker_queue()->Enqueue(
-      [shared_client, user_promise, settings] {
-        User user = user_promise->get_future().get();
-        shared_client->Initialize(user, settings);
-      });
+  HARD_ASSERT(
+      credentials_initialized,
```

Comment:
Now that `credentials_changed` is captured by reference it's no longer value to use in the `credential_change_listener` once it escapes the current stack frame. Let's move `credentials_listener` to be a member of `FirestoreClient` to avoid these lifetime issues.

Reply:
Fun. This is fixed.